src/app/redux/TransactionReducer.js

```diff
@@ -0,0 +1,123 @@
+                        console.error('You already voted for this post.')
+                    }
+                    break;
+                case 'comment':
+                    if (/You may only post once per minute/.test(errorStr)) {
+                        errorKey = 'You may only post once per minute.'
+                    } else if (errorStr === 'Testing, fake error')
+                        errorKey = 'Testing, fake error';
+                    break;
+                case 'transfer':
+                    if (/get_balance/.test(errorStr)) {
+                        errorKey = 'Insufficient balance.'
+                    }
+                    break;
+                case 'withdraw_vesting':
+                    if(/Account registered by another account requires 10x account creation fee worth of Steem Power/.test(errorStr))
+                        errorKey = 'Account requires 10x the account creation fee in Steem Power (approximately 300 SP) before it can power down.'
+                    break;
+                default:
+                    break;
+            }
+            if (state.hasIn(['TransactionError', type + '_listener'])) {
+                state = state.setIn(['TransactionError', type], fromJS({key: errorKey, exception: errorStr}))
+            } else {
+                if (error.message) {
+                    // Depends on FC_ASSERT formatting
+                    // https://github.com/steemit/steemit.com/issues/222
+                    const err_lines = error.message.split('\n');
+                    if (err_lines.length > 2) {
+                        errorKey = err_lines[1];
+                        const txt = errorKey.split(': ');
+                        if(txt.length && txt[txt.length - 1].trim() !== '') {
+                            errorKey = errorStr = txt[txt.length - 1]
+                        } else
+                            errorStr = `Transaction failed: ${err_lines[1]}`;
+                    }
+                }
+                if (errorStr.length > 200) errorStr = errorStr.substring(0, 200);
+                // Catch for unknown key better error handling
+                if (/unknown key: /.test(errorKey)) {
+                    errorKey = "Steem account doesn't exist.";
+                    errorStr = "Transaction failed: Steem account doesn't exist.";
+                }
+                // Catch for invalid active authority
+                if (/Missing Active Authority /.test(errorKey)) {
+                    errorKey = "Not your valid active key.";
+                    errorStr = "Transaction failed: Not your valid active key.";
+                }
+                state = state.update('errors', errors => {
+                    return errors ? errors.set(errorKey, errorStr) : Map({[errorKey]: errorStr});
+                });
+            }
+        }
+        if (errorCallback) try { errorCallback(errorKey) } catch (error2) { console.error(error2) }
```

Comment:
1. always brace conditional statements.
2. Do not hide such a massive, consequential application behavior by removing newlines.
3. do not `On Error Resume Next`. If there is an error it should blow up the application in a suitably panicky way.

Reply:
again this is a port of the previously defined reducer. i can work on refactoring it now if you think it's a good idea.

Reply:
```js
if (errorCallback) {
  errorCallback(errorKey);
} else {
  throw new Error('PANIC: no callback registered to handle error ' + errorKey);
}
```

that was fast

Reply:
looks like the original behavior is old as dirt https://github.com/steemit/condenser/blame/master/src/app/redux/Transaction.js#L113

i'm not super familiar with how this part of the app works, and would like others' feedback first